apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -16,36 +22,56 @@ import {
 } from "./styles";
 import { useFilters } from "./useFilters";
 import { useWaterfallTrace } from "./useWaterfallTrace";
+import { groupInactiveVersions } from "./utils";
 import { VersionLabel, VersionLabelView } from "./VersionLabel";
 
 type WaterfallGridProps = {
   projectIdentifier: string;
+  setPagination: (pagination: WaterfallPagination) => void;
 };
 
 export const WaterfallGrid: React.FC<WaterfallGridProps> = ({
   projectIdentifier,
+  setPagination,
 }) => {
   useWaterfallTrace();
 
+  const [maxOrder] = useQueryParam<number>(WaterfallFilterOptions.MaxOrder, 0);
+  const [minOrder] = useQueryParam<number>(WaterfallFilterOptions.MinOrder, 0);
+
   const { data } = useSuspenseQuery<WaterfallQuery, WaterfallQueryVariables>(
     WATERFALL,
     {
       variables: {
         options: {
           projectIdentifier,
           limit: VERSION_LIMIT,
+          maxOrder,
+          minOrder,
         },
       },
       // @ts-expect-error pollInterval isn't officially supported by useSuspenseQuery, but it works so let's use it anyway.
       pollInterval: DEFAULT_POLL_INTERVAL,
     },
   );
+
+  useEffect(() => {
+    setPagination(data.waterfall.pagination);
+  }, [setPagination, data.waterfall.pagination]);
+
   const refEl = useRef<HTMLDivElement>(null);
   const { height } = useDimensions(
     refEl as React.MutableRefObject<HTMLElement>,
   );
 
-  const { buildVariants, versions } = useFilters(data.waterfall);
+  const groupedVersions = groupInactiveVersions(
+    data.waterfall.flattenedVersions,
+  );
```

Comment:
If we stick with flattened versions, would it make sense to do this in `useFilters`?